Cucumber-JVM v1.2.3 (and v1.2.4) fails to build for Android

As reported on the [mailing list](https://groups.google.com/forum/#!topic/cukes/tAwLqtWhjcI), Cucumber-JVM release v1.2.3 (and v1.2.4) fails to build for Android. The issue occurred when Cucumber-JVM-Deps was updated to use XStream v1.4.8, which contains two classes (com.thoughtworks.xstream.mapper.LambdaMapper.class and com.thoughtworks.xstream.converters.reflection.LambdaConverter.class) compiled to the byte code version of Java8 (major version 52), and classes of that byte code version cannot be included in an Android dex file.

cucumber/cucumber-jvm-deps#4 fixes the problem by filtering out those two classes from the Cucumber-JVM-Deps jar file.

What is the latest version that is known to work?

I was adding support for Cucumber to my existing android app. Added the following JAR files:

- cucumber-android-1.2.4.jar
- cucumber-core-1.2.4.jar
- cucumber-html-0.2.3.jar
- cucumber-java-1.2.4.jar
- cucumber-junit-1.2.4.jar
- cucumber-jvm-deps-1.0.5.jar
- gherkin-2.12.2.jar

Getting this error for "./gradlew --info clean assembleDebug":

```
processing cucumber/deps/com/thoughtworks/xstream/mapper/DynamicProxyMapper$DynamicProxy.class...
processing cucumber/deps/com/thoughtworks/xstream/mapper/PackageAliasingMapper$1.class...
processing cucumber/deps/com/thoughtworks/xstream/mapper/LambdaMapper.class...

Unknown source file : UNEXPECTED TOP-LEVEL EXCEPTION:
Unknown source file : com.android.dx.cf.iface.ParseException: bad class file magic (cafebabe) or version (0034.0000)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parse0(DirectClassFile.java:472)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parse(DirectClassFile.java:406)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.parseToInterfacesIfNecessary(DirectClassFile.java:388)
Unknown source file : 	at com.android.dx.cf.direct.DirectClassFile.getMagic(DirectClassFile.java:251)
Unknown source file : 	at com.android.dx.command.dexer.Main.processClass(Main.java:704)
Unknown source file : 	at com.android.dx.command.dexer.Main.processFileBytes(Main.java:673)
Unknown source file : 	at com.android.dx.command.dexer.Main.access$300(Main.java:83)
Unknown source file : 	at com.android.dx.command.dexer.Main$1.processFileBytes(Main.java:602)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.processArchive(ClassPathOpener.java:284)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.processOne(ClassPathOpener.java:166)
Unknown source file : 	at com.android.dx.cf.direct.ClassPathOpener.process(ClassPathOpener.java:144)
Unknown source file : 	at com.android.dx.command.dexer.Main.processOne(Main.java:632)
Unknown source file : 	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:510)
Unknown source file : 	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:280)
Unknown source file : 	at com.android.dx.command.dexer.Main.run(Main.java:246)
Unknown source file : 	at com.android.dx.command.dexer.Main.main(Main.java:215)
Unknown source file : 	at com.android.dx.command.Main.main(Main.java:106)
Unknown source file : ...while parsing cucumber/deps/com/thoughtworks/xstream/mapper/LambdaMapper.class
Unknown source file : 1 error; aborting

:app:preDexDebug FAILED
:app:preDexDebug (Thread[main,5,main]) completed. Took 7.305 secs.
Starting process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_05.jdk/Contents/Home/bin/java''. Working directory: /Users/leninraj/Dropbox/Projects/duggout/duggout-android/app Command: /Library/Java/JavaVirtualMachines/jdk1.8.0_05.jdk/Contents/Home/bin/java -Xmx1024M -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /Users/leninraj/Library/Android/sdk/build-tools/21.1.2/lib/dx.jar com.android.dx.command.Main --dex --verbose --output /Users/leninraj/Dropbox/Projects/duggout/duggout-android/app/build/intermediates/pre-dexed/debug/support-annotations-22.2.1-572ba533a1ff5ff170aeef85dc74e608e4256da3.jar /Users/leninraj/Library/Android/sdk/extras/android/m2repository/com/android/support/support-annotations/22.2.1/support-annotations-22.2.1.jar
Successfully started process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_05.jdk/Contents/Home/bin/java''

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDexDebug'.
> com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_05.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output.

BUILD FAILED

Total time: 22.186 secs
PREDEX CACHE HITS:   0
PREDEX CACHE MISSES: 7
Stopped 0 compiler daemon(s).
```

If I remove "cucumber-jvm-deps-1.0.5.jar" from the libs directory, build works correctly. Which JAR file should I downgrade to get cucumber working for the Android App?

--- a/clojure/pom.xml
+++ b/clojure/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-html</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
         </dependency>
         <dependency>

--- a/examples/clojure_cukes/pom.xml
+++ b/examples/clojure_cukes/pom.xml
@@ -44,7 +44,7 @@
             <artifactId>clojure</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/examples/groovy-calculator/pom.xml
+++ b/examples/groovy-calculator/pom.xml
@@ -17,7 +17,7 @@
 
     <dependencies>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/examples/java-calculator-testng/pom.xml
+++ b/examples/java-calculator-testng/pom.xml
@@ -13,7 +13,7 @@
 
     <dependencies>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/examples/java-calculator/pom.xml
+++ b/examples/java-calculator/pom.xml
@@ -13,7 +13,7 @@
 
     <dependencies>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/examples/pax-exam/calculator-test/src/test/java/cucumber/examples/java/paxexam/test/CalculatorTest.java
+++ b/examples/pax-exam/calculator-test/src/test/java/cucumber/examples/java/paxexam/test/CalculatorTest.java
@@ -54,7 +54,7 @@ public class CalculatorTest {
 
             mavenBundle("io.cucumber", "gherkin"),
             mavenBundle("io.cucumber", "tag-expressions"),
-            mavenBundle("info.cukes", "cucumber-jvm-deps"),
+            mavenBundle("io.cucumber", "cucumber-jvm-deps"),
             mavenBundle("io.cucumber", "cucumber-core"),
             mavenBundle("io.cucumber", "cucumber-java"),
             mavenBundle("io.cucumber", "cucumber-osgi"),

--- a/examples/scala-calculator/pom.xml
+++ b/examples/scala-calculator/pom.xml
@@ -13,7 +13,7 @@
 
     <dependencies>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/examples/spring-txn/pom.xml
+++ b/examples/spring-txn/pom.xml
@@ -79,7 +79,7 @@
         </dependency>
 
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>test</scope>
         </dependency>

--- a/groovy/pom.xml
+++ b/groovy/pom.xml
@@ -22,7 +22,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
         </dependency>
         <dependency>

--- a/guice/pom.xml
+++ b/guice/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/jruby/pom.xml
+++ b/jruby/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
         </dependency>
         <dependency>

--- a/junit/pom.xml
+++ b/junit/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/jython/pom.xml
+++ b/jython/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
         </dependency>
         <dependency>

--- a/needle/pom.xml
+++ b/needle/pom.xml
@@ -22,7 +22,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/openejb/pom.xml
+++ b/openejb/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/osgi/pom.xml
+++ b/osgi/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/picocontainer/pom.xml
+++ b/picocontainer/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
         </dependency>
         <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -95,9 +95,9 @@
     <dependencyManagement>
         <dependencies>
             <dependency>
-                <groupId>info.cukes</groupId>
+                <groupId>io.cucumber</groupId>
                 <artifactId>cucumber-jvm-deps</artifactId>
-                <version>1.0.5</version>
+                <version>1.0.6</version>
                 <exclusions>
                     <exclusion>
                         <groupId>com.thoughtworks.xstream</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -113,6 +113,12 @@
                 <groupId>io.cucumber</groupId>
                 <artifactId>gherkin</artifactId>
                 <version>${gherkin.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>io.cucumber</groupId>
+                        <artifactId>gherkin-jvm-deps</artifactId>
+                    </exclusion>
+                </exclusions>
             </dependency>
             <dependency>
                 <groupId>io.cucumber</groupId>

--- a/rhino/pom.xml
+++ b/rhino/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/scala/pom.xml
+++ b/scala/pom.xml
@@ -37,7 +37,7 @@
             <artifactId>cucumber-core</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/spring/pom.xml
+++ b/spring/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

--- a/weld/pom.xml
+++ b/weld/pom.xml
@@ -18,7 +18,7 @@
             <artifactId>cucumber-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>info.cukes</groupId>
+            <groupId>io.cucumber</groupId>
             <artifactId>cucumber-jvm-deps</artifactId>
             <scope>provided</scope>
         </dependency>

